iceoryx_hoofs/test/moduletests/test_relative_pointer.cpp

```diff
@@ -308,138 +233,120 @@ TYPED_TEST(relativeptrtests, AssignmentOperatorTests)
 }
 
-TYPED_TEST(relativeptrtests, unRegisterPointerTest_Valid)
+TYPED_TEST(base_relative_ptr_test, unRegisterPointerTest_Valid)
 {
-    MemMap memMap(this->m_fileDescriptor);
-    RelativePointer<TypeParam> rp1(memMap.getMappedAddress(), 1);
+    RelativePointer<TypeParam> rp1(this->memoryPartition[0], 1);
 
-    rp1.registerPtr(1, memMap.getMappedAddress());
+    rp1.registerPtr(1, this->memoryPartition[0]);
     EXPECT_EQ(rp1.unregisterPtr(1), true);
-    EXPECT_EQ(rp1.registerPtr(1, memMap.getMappedAddress()), true);
+    EXPECT_EQ(rp1.registerPtr(1, this->memoryPartition[0]), true);
 }
 
-TYPED_TEST(relativeptrtests, unregisterPointerAll)
+TYPED_TEST(base_relative_ptr_test, unregisterPointerAll)
 {
-    MemMap memMap(this->m_fileDescriptor);
-    MemMap memMap1(this->m_fileDescriptor);
+    RelativePointer<TypeParam> rp1(this->memoryPartition[0], 1);
+    RelativePointer<TypeParam> rp2(this->memoryPartition[1], 9999);
 
-    RelativePointer<TypeParam> rp1(memMap.getMappedAddress(), 1);
-    RelativePointer<TypeParam> rp2(memMap1.getMappedAddress(), 9999);
-
-    EXPECT_EQ(rp1.registerPtr(1, memMap.getMappedAddress()), true);
-    EXPECT_EQ(rp2.registerPtr(9999, memMap1.getMappedAddress()), true);
+    EXPECT_EQ(rp1.registerPtr(1, this->memoryPartition[0]), true);
+    EXPECT_EQ(rp2.registerPtr(9999, this->memoryPartition[1]), true);
     BaseRelativePointer::unregisterAll();
-    EXPECT_EQ(rp1.registerPtr(1, memMap.getMappedAddress()), true);
-    EXPECT_EQ(rp2.registerPtr(9999, memMap1.getMappedAddress()), true);
+    EXPECT_EQ(rp1.registerPtr(1, this->memoryPartition[0]), true);
+    EXPECT_EQ(rp2.registerPtr(9999, this->memoryPartition[1]), true);
 }
 
-TYPED_TEST(relativeptrtests, registerPtrWithId)
+TYPED_TEST(base_relative_ptr_test, registerPtrWithId)
 {
-    MemMap memMap(this->m_fileDescriptor);
-    MemMap memMap1(this->m_fileDescriptor);
-
-    RelativePointer<TypeParam> rp1(memMap.getMappedAddress(), 1);
-    RelativePointer<TypeParam> rp2(memMap1.getMappedAddress(), 10000);
+    RelativePointer<TypeParam> rp1(this->memoryPartition[0], 1);
+    RelativePointer<TypeParam> rp2(this->memoryPartition[1], 10000);
 
-    EXPECT_EQ(rp1.registerPtr(1, memMap.getMappedAddress()), true);
-    EXPECT_EQ(rp2.registerPtr(10000, memMap1.getMappedAddress()), false);
+    EXPECT_EQ(rp1.registerPtr(1, this->memoryPartition[0]), true);
+    EXPECT_EQ(rp2.registerPtr(10000, this->memoryPartition[1]), false);
 }
 
-TYPED_TEST(relativeptrtests, basePointerValid)
+TYPED_TEST(base_relative_ptr_test, basePointerValid)
 {
-    MemMap memMap(this->m_fileDescriptor);
-    void* basePtr1 = memMap.getMappedAddress();
+    void* basePtr1 = this->memoryPartition[0];
 
-    RelativePointer<TypeParam> rp1(memMap.getMappedAddress(), 1);
+    RelativePointer<TypeParam> rp1(this->memoryPartition[0], 1);
     EXPECT_EQ(rp1.getBasePtr(1), nullptr);
-    rp1.registerPtr(1, memMap.getMappedAddress());
+    rp1.registerPtr(1, this->memoryPartition[0]);
```

Comment:
similarly here, the original tests uses
```
static bool registerPtr(const id_t id, const ptr_t ptr, uint64_t size = 0U) noexcept;
```
you use
```
static id_t registerPtr(const ptr_t ptr, uint64_t size = 0U) noexcept;
```
This will break once you initialize the array with a dummy pattern.

On a side note, the methods should not have the same name to prevent this easy to make errors.

Reply:
Actually this is the same signature since `memoryPartition` is defined like: `uint8_t memoryPartition[NUMBER_OF_MEMORY_PARTITIONS][SHARED_MEMORY_SIZE];` therefore `this->memoryPartition[0]` provides my a pointer to the first element of the second array.